rebin_dataset.py

```diff
@@ -0,0 +1,86 @@
+from cpyImagingMSpec import ImzbReader
+import argparse
+import dask.array as da
+import numpy as np
+from toolz import partition_all
+from mz_axis import generate_mz_axis, Instrument
+
+
+def get_mz_images(mz_axis_chunk, imzb):
+    imgs = np.zeros((len(mz_axis_chunk), imzb.height, imzb.width))
+    for n, (mz, ppm) in enumerate(mz_axis_chunk):
+        img = imzb.get_mz_image(mz, ppm)
+        img[img < 0] = 0
+        perc = np.percentile(img, 99)
+        img[img > perc] = perc
+        imgs[n, :, :] = img
+    return imgs
+
+
+def open_dataset(input_filename):
+    imzb = ImzbReader(input_filename)
+    return imzb
+
+
+def get_mz_range(imzb):
+    mz_range = (imzb.min_mz - 0.1, imzb.max_mz + 0.1)
+    assert mz_range[0] > 0
+    assert mz_range[1] > 1
+    return mz_range
+
+
+def get_mz_axis(min_mz, max_mz, instrument, step=1.0):
+    mz_axis = generate_mz_axis(min_mz, max_mz, instrument, step)
+    return mz_axis
+
+
+def save_image_intensities(image_intensities, output_filename, instrument, mz_axis):
+    with open(output_filename, "w+") as f:
+        np.savez_compressed(f, image_intensities=image_intensities, instrument=instrument, mz_axis=mz_axis)
+
+
+def parallel_rebin(K, mz_axis, imzb):
+    mz_axis_chunks = list(partition_all(K, mz_axis))
+    # create dask array manually using tasks
+    tasks = {('x', i, 0, 0): (get_mz_images, mz_chunk, imzb) for i, mz_chunk in enumerate(mz_axis_chunks)}
+    chunks_mz = [len(c) for c in mz_axis_chunks]
+    chunks_x = (imzb.height,)
+    chunks_y = (imzb.width,)
+    arr = da.Array(tasks, 'x', chunks=(chunks_mz, chunks_x, chunks_y), dtype=float)
+    print arr.shape
+    return arr
+
+
+def get_instrument_model(instrument_type, res200):
+    class hackalot():
```

Comment:
that's horrible, please modify `Instrument` constructor instead to take two parameters